packages/web/src/components/payment-method/PaymentMethod.tsx

```diff
@@ -0,0 +1,168 @@
+import { ChangeEvent, useCallback } from 'react'
+
+import {
+  BNUSDC,
+  Nullable,
+  PurchaseMethod,
+  PurchaseVendor,
+  formatCurrencyBalance,
+  formatUSDCWeiToFloorCentsNumber
+} from '@audius/common'
+import {
+  FilterButton,
+  FilterButtonType,
+  Flex,
+  IconCreditCard,
+  IconDonate,
+  IconTransaction
+} from '@audius/harmony'
+import { RadioButton, RadioButtonGroup } from '@audius/stems'
+import BN from 'bn.js'
+
+import { MobileFilterButton } from 'components/mobile-filter-button/MobileFilterButton'
+import { SummaryTable, SummaryTableItem } from 'components/summary-table'
+import { Text } from 'components/typography'
+import { isMobile } from 'utils/clientUtil'
+import zIndex from 'utils/zIndex'
+
+const messages = {
+  paymentMethod: 'Payment Method',
+  withExistingBalance: 'Existing balance',
+  withCard: 'Checkout with card',
+  withCrypto: 'Add funds with crypto transfer'
+}
+
+type PaymentMethodProps = {
+  selectedType: Nullable<PurchaseMethod>
+  setSelectedType: (method: PurchaseMethod) => void
+  balance?: Nullable<BNUSDC>
+  isExistingBalanceDisabled?: boolean
+  showExistingBalance?: boolean
+  cardMessage?: string
+  cryptoMessage?: string
+}
+
+export const PaymentMethod = ({
+  selectedType,
+  setSelectedType,
+  balance,
+  isExistingBalanceDisabled,
+  showExistingBalance,
+  cardMessage,
+  cryptoMessage
+}: PaymentMethodProps) => {
+  const mobile = isMobile()
+  const balanceCents = formatUSDCWeiToFloorCentsNumber(
+    (balance ?? new BN(0)) as BNUSDC
+  )
+  const balanceFormatted = formatCurrencyBalance(balanceCents / 100)
+  const vendorOptions = [{ label: PurchaseVendor.STRIPE }]
+
+  const options = [
+    showExistingBalance
+      ? {
+          id: PurchaseMethod.BALANCE,
+          label: messages.withExistingBalance,
+          icon: IconDonate,
+          disabled: isExistingBalanceDisabled,
+          value: (
+            <Text
+              as='span' // Needed to avoid <p> inside <p> warning
+              variant='title'
+              color={
+                selectedType === PurchaseMethod.BALANCE
+                  ? 'secondary'
+                  : undefined
+              }
+            >
+              ${balanceFormatted}
+            </Text>
+          )
+        }
+      : null,
+    {
+      id: PurchaseMethod.CARD,
+      label: cardMessage || messages.withCard,
+      icon: IconCreditCard,
+      value:
+        vendorOptions.length > 1 ? (
+          mobile ? (
+            <MobileFilterButton
```

Comment:
can you try adding another vendor option and see if mobile web looks correct? iirc we may need to switch flex-direction to column on mobile web to stack the filter button underneath the title.

Reply:
good catch! resolved